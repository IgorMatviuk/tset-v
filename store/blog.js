export const mutationTypes = {
  SET_LOADING_START: 'SET_LOADING_START',
  SET_LOADING_SUCCESS: 'SET_LOADING_SUCCESS',
  SET_CURRENT_PAGE: 'SET_CURRENT_PAGE'
}

export const actionsTypes = {
  GET_BLOG_POSTS: 'GET_BLOG_POSTS'
}

export const state = () => ({
  posts: [],
  isLoading: false,
  currentPage: null,
  totalPages: null
})

export const mutations = {
  [mutationTypes.SET_LOADING_START](state, status) {
    state.isLoading = status
  },
  [mutationTypes.SET_CURRENT_PAGE](state, currentPage) {
    state.currentPage = currentPage
  },
  [mutationTypes.SET_LOADING_SUCCESS](
    state,
    { posts, currentPage, totalPages }
  ) {
    state.posts = posts
    state.currentPage = currentPage
    state.totalPages = totalPages

    if (state.isLoading) {
      state.isLoading = false
    }
  }
}

export const actions = {
  async [actionsTypes.GET_BLOG_POSTS]({ commit }, params) {
    commit(mutationTypes.SET_LOADING_START, true)

    let query = Object.keys(params)
      .map((key) => `${key}=${encodeURI(params[key])}`)
      .join('&')

    if (!params.limit) {
      query += '&limit=8'
    }

    const { data, meta } = await this.$axios.$get(`api/posts?${query}`)

    const totalPages = Math.ceil(meta.total / (params.limit || 8))

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      posts: data,
      currentPage: meta.current_page,
      totalPages
    })
  }
}
