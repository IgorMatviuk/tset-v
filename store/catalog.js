import { CATALOG_ITEMS_PER_PAGE } from '@/utils/constants'

function createRootCategories(categories) {
  const withRootList = categories.map((category) => {
    if (category.name_alias) {
      const rootCategory = {
        id: category.id,
        name: category.name_alias,
        slug: category.slug
      }

      category.categories = [
        rootCategory,
        ...createRootCategories(category.categories)
      ]
    }

    return category
  })

  return withRootList
}

export const mutationTypes = {
  SET_CATEGORIES: 'SET_CATEGORIES',
  SET_ACTIVE_CATEGORY: 'SET_ACTIVE_CATEGORY',
  SET_PRODUCTS: 'SET_PRODUCTS',
  SET_ATTRIBUTES: 'SET_ATTRIBUTES',
  SET_FILTERS: 'SET_FILTERS',
  CLEAR_FILTERS: 'CLEAR_FILTERS',
  SET_CATALOG_DATA: 'SET_CATALOG_DATA',
  SET_ORDER: 'SET_ORDER',
  SET_SHOW_ITEMS_COUNT: 'SET_SHOW_ITEMS_COUNT',
  SET_LOADING_TOGGLE: 'SET_LOADING_TOGGLE',
  SET_CURRENT_PAGE: 'SET_CURRENT_PAGE',
  SET_TOTAL_PAGES: 'SET_TOTAL_PAGES',
  RESET_META_DATA: 'RESET_META_DATA',
  SET_SEO_TEXT: 'SET_SEO_TEXT'
}

export const actionTypes = {
  GET_CATEGORIES: 'GET_CATEGORIES',
  GET_PRODUCTS: 'GET_PRODUCTS',
  GET_ATTRIBUTES: 'GET_ATTRIBUTES'
}

export const state = () => ({
  categories: [],
  activeCategory: null,
  products: [],
  attributes: [],
  filters: [],
  order: 'popularity',
  showItemsCount: 4,
  isLoading: false,
  currentPage: null,
  totalPages: null,
  seoText: ''
})

export const mutations = {
  [mutationTypes.SET_CATEGORIES](state, categories) {
    state.categories = categories
  },
  [mutationTypes.SET_ACTIVE_CATEGORY](state, activeCategory) {
    state.activeCategory = activeCategory
  },
  [mutationTypes.SET_PRODUCTS](state, products) {
    state.products = products
  },
  [mutationTypes.SET_ATTRIBUTES](state, attributes) {
    state.attributes = attributes
  },
  [mutationTypes.SET_CATALOG_DATA](state, { category, products, attributes }) {
    state.activeCategory = category
    state.products = products
    state.attributes = attributes
  },
  [mutationTypes.SET_ORDER](state, order) {
    state.order = order
  },
  [mutationTypes.SET_SHOW_ITEMS_COUNT](state, showItemsCount) {
    state.showItemsCount = showItemsCount
  },
  [mutationTypes.SET_FILTERS](state, filters) {
    state.filters = filters
  },
  [mutationTypes.CLEAR_FILTERS](state) {
    state.order = 'popularity'
    state.filters = []
  },
  [mutationTypes.SET_LOADING_TOGGLE](state, status) {
    state.isLoading = status
  },
  [mutationTypes.SET_CURRENT_PAGE](state, currentPage) {
    state.currentPage = currentPage
  },
  [mutationTypes.SET_TOTAL_PAGES](state, totalPages) {
    state.totalPages = totalPages
  },
  [mutationTypes.RESET_META_DATA](state) {
    state.filters = []
    state.order = 'popularity'
    state.currentPage = 1
  },
  [mutationTypes.SET_SEO_TEXT](state, seoText) {
    state.seoText = seoText
  }
}

export const actions = {
  async [actionTypes.GET_CATEGORIES]({ commit }) {
    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: categories } = await this.$axios.$get(
      `${STAGE_API_URL}/api/categories`
    )

    // const { data: categories } = await this.$axios.$get('/api/categories')
    const normilizedCategories = createRootCategories(categories)

    commit(mutationTypes.SET_CATEGORIES, normilizedCategories)
  },
  async [actionTypes.GET_PRODUCTS]({ commit }, { url, query }) {
    commit(mutationTypes.SET_LOADING_TOGGLE, true)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const {
      data: { products },
      meta
    } = await this.$axios.$get(
      `${STAGE_API_URL}/api/router?url=${url}${query ? `&${query}` : ''}`
    )

    const totalPages = Math.ceil(meta.total / CATALOG_ITEMS_PER_PAGE)

    commit(mutationTypes.SET_PRODUCTS, products)
    commit(mutationTypes.SET_CURRENT_PAGE, meta.current_page)
    commit(mutationTypes.SET_TOTAL_PAGES, totalPages)
    commit(mutationTypes.SET_LOADING_TOGGLE, false)
  }
}

export const getters = {
  defaultCategorySlug(state) {
    // FIXME: temporary solution
    return state.categories[0].slug
  },
  optionLabel(state) {
    const { attributes } = state

    return (code, slug) => {
      const { options } = attributes.find((attr) => attr.code === code)
      const { label } = options.find((option) => option.slug === slug)
      return label
    }
  },
  selectedAttributeOptions(state) {
    const { filters } = state

    return (code) => {
      const filter = filters.find((filter) => filter.code === code)
      return filter ? filter.value : null
    }
  }
}
