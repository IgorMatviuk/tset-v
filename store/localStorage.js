export const mutationTypes = {
  SET_CART_ID: 'SET_CART_ID'
}

export const state = () => ({
  cartId: null,
  status: false
})

export const mutations = {
  [mutationTypes.SET_CART_ID](state, cartId) {
    state.cartId = cartId
  }
}
