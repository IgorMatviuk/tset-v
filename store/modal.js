export const mutationTypes = {
  SET_MODAL_META: 'SET_MODAL_META',
  SET_MODAL_VISIBILITY_TOGGLE: 'SET_MODAL_VISIBILITY_TOGGLE',
  SET_MODAL_DATA: 'SET_MODAL_DATA',
  RESET_MODAL_STATE: 'RESET_MODAL_STATE'
}

function createInitialState() {
  return {
    form: {
      visible: false,
      meta: {
        name: '',
        title: '',
        fields: [],
        actionText: '',
        submitText: '',
        links: [],
        action: {}
      },
      data: {}
    },
    message: {
      visible: false,
      meta: {
        title: '',
        content: ''
      }
    },
    testimonials: {
      visible: false,
      data: {}
    }
  }
}

export const state = () => createInitialState()

export const mutations = {
  [mutationTypes.SET_MODAL_META](state, payload) {
    const { type, ...meta } = payload

    if (type === 'form' && meta.success) {
      state.message.meta = meta.success
    }

    state[type].meta = meta
  },
  [mutationTypes.SET_MODAL_VISIBILITY_TOGGLE](state, { type, status }) {
    state[type].visible = status
  },
  [mutationTypes.SET_MODAL_DATA](state, { type, data }) {
    state[type].data = data
  },
  [mutationTypes.RESET_MODAL_STATE](state) {
    const { form, message, testimonials } = createInitialState()

    state.form = form
    state.message = message
    state.testimonials = testimonials
  }
}
