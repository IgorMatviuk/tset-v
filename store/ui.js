export const mutationTypes = {
  SET_PRELOADER_VISIBILITY_TOGGLE: 'SET_PRELOADER_VISIBILITY_TOGGLE',
  SET_BACKGROUND_REVEALED_TOGGLE: 'SET_BACKGROUND_REVEALED_TOGGLE',
  SET_CATEGORIES_DRAWER_VISIBILITY_TOGGLE:
    'SET_CATEGORIES_DRAWER_VISIBILITY_TOGGLE',
  SET_FILTERS_DRAWER_VISIBILITY_TOGGLE: 'SET_FILTERS_DRAWER_VISIBILITY_TOGGLE',
  SET_FLOAT_PANEL_VISIBILITY_TOGGLE: 'SET_FLOAT_PANEL_VISIBILITY_TOGGLE',
  SET_TERMS_OF_USE_VISIBILITY_TOGGLE: 'SET_TERMS_OF_USE_VISIBILITY_TOGGLE',
  SET_MOBILE_MENU_VISIBILITY_TOGGLE: 'SET_MOBILE_MENU_VISIBILITY_TOGGLE'
}

export const state = () => {
  return {
    isPreloaderVisible: true,
    isBackgroundRevealed: false,
    isCategoriesDrawerVisible: false,
    isFiltersDrawerVisible: false,
    isFloatPanelActive: false,
    isTermsOfUseVisible: false,
    isMobileMenuVisible: false
  }
}

export const mutations = {
  [mutationTypes.SET_PRELOADER_VISIBILITY_TOGGLE](state, status) {
    state.isPreloaderVisible = status
  },
  [mutationTypes.SET_BACKGROUND_REVEALED_TOGGLE](state, status) {
    state.isBackgroundRevealed = status
  },
  [mutationTypes.SET_CATEGORIES_DRAWER_VISIBILITY_TOGGLE](state, status) {
    state.isCategoriesDrawerVisible = status
  },
  [mutationTypes.SET_FILTERS_DRAWER_VISIBILITY_TOGGLE](state, status) {
    state.isFiltersDrawerVisible = status
  },
  [mutationTypes.SET_FLOAT_PANEL_VISIBILITY_TOGGLE](state, status) {
    state.isFloatPanelActive = status
  },
  [mutationTypes.SET_TERMS_OF_USE_VISIBILITY_TOGGLE](state, status) {
    state.isTermsOfUseVisible = status
  },
  [mutationTypes.SET_MOBILE_MENU_VISIBILITY_TOGGLE](state, status) {
    state.isMobileMenuVisible = status
  }
}
