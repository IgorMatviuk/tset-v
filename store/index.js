import { actionTypes as catalogActionTypes } from '@/store/catalog'

export const mutationTypes = {
  SET_LANG: 'SET_LANG',
  SET_MENUS: 'SET_MENUS',
  SET_PAGE_TYPE: 'SET_PAGE_TYPE',
  SET_BREADCRUMB: 'SET_BREADCRUMB',
  SET_CONTACTS: 'SET_CONTACTS',
  SET_GLOBALS: 'SET_GLOBALS'
}

export const actionTypes = {
  GET_MENUS: 'GET_MENUS',
  GET_CONTACTS: 'GET_CONTACTS',
  GET_BANNER: 'GET_BANNER',
  GET_CLIENT_LOCATION: 'GET_CLIENT_LOCATION'
}

export const state = () => {
  return {
    locale: 'ru',
    locales: ['uk', 'ru'],
    fallbackLocale: 'ru',
    pageType: null,
    breadcrumb: [],
    menus: [],
    globals: {
      countryCode: 'UA',
      banner: null
    },
    contacts: {
      phone: '0 800 332 512',
      email: 'office@virna.com.ua',
      schedule: 'с 9:00 - 18:00',
      socialLinks: [
        { name: 'instagram', url: 'https://www.instagram.com/' },
        { name: 'facebook', url: 'https://facebook.com/' },
        { name: 'youtube', url: 'https://www.youtube.com/' }
      ]
    }
  }
}

export const mutations = {
  [mutationTypes.SET_LANG](state, locale) {
    state.locale = locale
  },
  [mutationTypes.SET_MENUS](state, menus) {
    state.menus = menus
  },
  [mutationTypes.SET_PAGE_TYPE](state, pageType) {
    state.pageType = pageType
  },
  [mutationTypes.SET_BREADCRUMB](state, breadcrumb) {
    state.breadcrumb = breadcrumb
  },
  [mutationTypes.SET_CONTACTS](state, contacts) {
    state.contacts = contacts
  },
  [mutationTypes.SET_GLOBALS](state, { key, value }) {
    state.globals[key] = value
  }
}

export const actions = {
  async [actionTypes.GET_MENUS]({ commit }) {
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: menus } = await this.$axios.$get(`${STAGE_API_URL}/api/menus`)
    commit(mutationTypes.SET_MENUS, menus)
  },
  async [actionTypes.GET_CONTACTS]({ commit }) {
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: contacts } = await this.$axios.$get(
      `${STAGE_API_URL}/api/contacts`
    )
    commit(mutationTypes.SET_CONTACTS, contacts)
  },
  async [actionTypes.GET_BANNER]({ commit }) {
    const { data: banner } = await this.$axios.$get('/api/banner')
    const defaultBanner = {
      key: 'banner',
      value: banner
    }

    commit(mutationTypes.SET_GLOBALS, defaultBanner)
  },
  async [actionTypes.GET_CLIENT_LOCATION]({ commit }) {
    const locationString = await this.$axios.$get('https://ip2c.org/s')
    const [, countryCode] = locationString.split(';')
    const defaultCountry = {
      key: 'countryCode',
      value: countryCode
    }

    commit(mutationTypes.SET_GLOBALS, defaultCountry)
  },
  async nuxtServerInit({ dispatch }) {
    await Promise.all([
      dispatch(actionTypes.GET_MENUS),
      dispatch(actionTypes.GET_CONTACTS),
      dispatch(`catalog/${catalogActionTypes.GET_CATEGORIES}`)
    ])
  }
}

/**
 * Get menum by type
 * @param {object} state
 * @param {('header'|'footer'|'catalog'|'promo')} type
 */
function getMenuByType(state, type) {
  const { menus } = state
  return menus.find((menu) => menu.menu_type === type)
}

export const getters = {
  mobileMenu(state) {
    return getMenuByType(state, 'mobile')
  },
  desktopMenu(state) {
    return getMenuByType(state, 'header')
  },
  promoMenu(state) {
    return getMenuByType(state, 'promo')
  },
  topMenu(state) {
    return getMenuByType(state, 'top')
  },
  catalogMenu(state) {
    return getMenuByType(state, 'catalog')
  },
  footerMenu(state) {
    return getMenuByType(state, 'footer')
  }
}
