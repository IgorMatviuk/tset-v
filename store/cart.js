import { mutationTypes as localStorageMutationTypes } from '@/store/localStorage'

export const mutationTypes = {
  SET_LOADING_START: 'SET_LOADING_START',
  SET_LOADING_SUCCESS: 'SET_LOADING_SUCCESS',
  SET_CART_DATA: 'SET_CART_DATA',
  SET_COUNTRIES: 'SET_COUNTRIES',
  SET_REGIONS: 'SET_REGIONS',
  SET_STATES: 'SET_STATES',
  SET_ORDER_SAVE: 'SET_ORDER_SAVE',
  CLEAR_CART: 'CLEAR_CART'
}

export const actionTypes = {
  GET_CHECKOUT_DATA: 'GET_CHECKOUT_DATA',
  GET_CART: 'GET_CART',
  GET_COUNTRIES: 'GET_COUNTRIES',
  GET_REGIONS: 'GET_REGIONS',
  GET_STATES: 'GET_STATES',
  GET_OFFICES: 'GET_OFFICES',
  ADD_ITEM: 'ADD_ITEM',
  UPDATE_ITEM: 'UPDATE_ITEM',
  REMOVE_ITEM: 'REMOVE_ITEM'
}

export const state = () => ({
  cartData: {},
  countries: [],
  regions: [],
  states: [],
  offices: [],
  isOrderSaved: false,
  isLoading: false
})

export const mutations = {
  [mutationTypes.SET_CART_DATA](state, cartData) {
    state.cartData = cartData
  },
  [mutationTypes.SET_LOADING_START](state) {
    state.isLoading = true
  },
  [mutationTypes.SET_LOADING_SUCCESS](state, { key, value }) {
    if (state.isLoading) {
      state.isLoading = false
    }

    state[key] = value
  },
  [mutationTypes.SET_COUNTRIES](state, countries) {
    state.countries = countries
  },
  [mutationTypes.SET_REGIONS](state, regions) {
    state.regions = regions
  },
  [mutationTypes.SET_STATES](state, states) {
    state.states = states
  },
  [mutationTypes.SET_OFFICES](state, offices) {
    state.offices = offices
  },
  [mutationTypes.SET_ORDER_SAVE](state) {
    state.isOrderSaved = true
    state.cartData = {}
  },
  [mutationTypes.CLEAR_CART](state) {
    state.cartData = {}
    state.isOrderSaved = false
  }
}

export const actions = {
  async [actionTypes.GET_CHECKOUT_DATA]({ commit }) {
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: checkoutData } = await this.$axios.$get(
      `${STAGE_API_URL}/api/checkout`
    )

    commit(mutationTypes.SET_COUNTRIES, checkoutData.countries)
    commit(mutationTypes.SET_REGIONS, checkoutData.regions)
  },
  async [actionTypes.GET_CART]({ commit }, cartId) {
    // const { data: cartData } = await this.$axios.$get('/api/checkout/cart')

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    // const STAGE_API_URL = 'http://localhost:7000'
    // const { data: cartData } = await this.$axios.$get(
    //   `${STAGE_API_URL}/api/checkout/cart`
    // )

    const { data: cartData } = await this.$axios.$get(
      `${STAGE_API_URL}/api/checkout/cart/${cartId}`
    )

    if (cartData) {
      commit(mutationTypes.SET_CART_DATA, cartData)
    }
  },
  async [actionTypes.GET_REGIONS]({ commit }, regionId) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: regions } = await this.$axios.$get(
      `${STAGE_API_URL}/api/shipping/regions`
    )

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'regions',
      value: regions
    })
  },
  async [actionTypes.GET_STATES]({ commit }, regionId) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: states } = await this.$axios.$get(
      `${STAGE_API_URL}/api/shipping/states?region_id=${regionId}`
    )

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'states',
      value: states
    })
  },
  async [actionTypes.GET_OFFICES]({ commit }, stateId) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: offices } = await this.$axios.$get(
      `${STAGE_API_URL}/api/shipping/offices?state_id=${stateId}`
    )

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'offices',
      value: offices
    })
  },
  async [actionTypes.ADD_ITEM]({ commit }, { cartId, itemId }) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    // const { data: cartData } = await this.$axios.$post(
    //   `/api/checkout/cart/add/${id}`
    // )

    const STAGE_API_URL = 'https://back.virna.com.ua'
    // const STAGE_API_URL = 'http://localhost:7000'
    const { data: cartData } = await this.$axios.$post(
      `${STAGE_API_URL}/api/checkout/cart/add`,
      {
        cart: cartId,
        variant: itemId
      }
    )

    if (!state.cartData) {
      commit(
        `localStorage/${localStorageMutationTypes.SET_CART_ID}`,
        cartData.id,
        { root: true }
      )
    }

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'cartData',
      value: cartData
    })
  },
  async [actionTypes.UPDATE_ITEM]({ commit }, { cartId, itemId, body }) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: cartData } = await this.$axios.$put(
      `${STAGE_API_URL}/api/checkout/cart/${cartId}/update/${itemId}`,
      body
    )

    // const { data: cartData } = await this.$axios.$put(
    //   `/api/checkout/cart/update/${id}`,
    //   body
    // )

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'cartData',
      value: cartData
    })
  },
  async [actionTypes.REMOVE_ITEM]({ commit }, { cartId, itemId }) {
    commit(mutationTypes.SET_LOADING_START)

    // FIXME: temporary solution
    const STAGE_API_URL = 'https://back.virna.com.ua'
    const { data: cartData } = await this.$axios.$delete(
      `${STAGE_API_URL}/api/checkout/cart/${cartId}/remove-item/${itemId}`
    )

    if (!cartData) {
      commit(`localStorage/${localStorageMutationTypes.SET_CART_ID}`, null, {
        root: true
      })
    }

    commit(mutationTypes.SET_LOADING_SUCCESS, {
      key: 'cartData',
      value: cartData || {}
    })
  }
}
