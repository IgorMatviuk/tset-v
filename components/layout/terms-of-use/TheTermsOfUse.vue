<template>
  <transition name="fade">
    <div v-if="isTermsOfUseVisible" class="terms-of-use">
      <div class="terms-of-use__content">
        <div class="terms-of-use__text">
          {{ $t('termsOfUse.content') }}
          <NuxtLink
            :to="$i18n.path('privacy-policy')"
            class="terms-of-use__link"
            @click.native="closeTermsOfUse"
          >
            {{ $t('termsOfUse.linkText') }}
          </NuxtLink>
        </div>
        <BButton
          variant="primary"
          class="terms-of-use__btn-accept"
          @click="closeTermsOfUse"
        >
          {{ $t('buttons.accept') }}
        </BButton>
        <button
          class="terms-of-use__close"
          type="button"
          @click="closeTermsOfUse"
        >
          <svg-icon
            name="close"
            class="terms-of-use__icon"
            width="14"
            height="14"
          />
        </button>
      </div>
    </div>
  </transition>
</template>

<script>
import { mutationTypes as uiMutationTypes } from '@/store/ui'

export default {
  computed: {
    isTermsOfUseVisible() {
      return this.$store.state.ui.isTermsOfUseVisible
    }
  },
  methods: {
    closeTermsOfUse() {
      this.$store.commit(
        `ui/${uiMutationTypes.SET_TERMS_OF_USE_VISIBILITY_TOGGLE}`,
        false
      )
    }
  }
}
</script>

<style lang="scss">
.terms-of-use {
  position: fixed;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 11;
  padding: 25px 0;
  background-color: $white;
  box-shadow: -2px -2px 12px rgba(41, 54, 136, 0.1);

  &__content {
    position: relative;
    max-width: 1200px;
    padding-right: 15px;
    padding-left: 15px;
    margin: 0 auto;
    font-size: 13px;
    line-height: 15px;
    color: $gray-dark-2;
  }

  &__link {
    color: $black;
    text-decoration: underline;

    @include hover-focus-active() {
      color: $black;
      text-decoration: underline;
    }
  }

  &__close {
    @include reset-button-styles();

    position: absolute;
    top: -25px;
    right: 0;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 40px;
    height: 40px;
    color: $gray-light-1;
    transition: color 0.3s;

    @include hover-focus-active() {
      color: $primary;
    }
  }

  &__icon {
    width: 14px;
    height: 14px;
  }

  @include media-breakpoint-down(lg) {
    padding: 15px 0;

    &__content {
      padding-right: 30px;
      padding-left: 15px;
      font-size: 12px;
      line-height: 14px;
    }

    &__close {
      top: -15px;
    }
  }

  @include media-breakpoint-down(sm) {
    &__btn-accept {
      min-width: 130px;
      height: 30px;
      margin-top: 15px;
      font-size: 12px;
      line-height: 28px;
    }
  }

  @include media-breakpoint-up(md) {
    &__content {
      display: flex;
      align-items: center;
      padding-right: 50px;
    }
  }

  @include media-breakpoint-up(xl) {
    &__content {
      padding-right: 60px;
    }

    &__text {
      padding-right: 30px;
    }
  }
}
</style>
