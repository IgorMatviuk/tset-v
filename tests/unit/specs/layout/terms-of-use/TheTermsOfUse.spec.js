import { createLocalVue, mount } from '@vue/test-utils'
import Vuex from 'vuex'
import { BButton } from 'bootstrap-vue'

import TheTermsOfUse from '@/components/layout/terms-of-use/TheTermsOfUse'

import { mutations as uiMutations } from '@/store/ui'

const localVue = createLocalVue()
localVue.component('BButton', BButton)
localVue.use(Vuex)

describe('TheImageViewer', () => {
  let store

  function createTheImageViewer() {
    const wrapper = mount(TheTermsOfUse, {
      localVue,
      store,
      mocks: {
        $t: (path) => path,
        $i18n: {
          path: (url) => url
        }
      },
      stubs: ['NuxtLink', 'svg-icon']
    })

    return wrapper
  }

  beforeEach(() => {
    store = new Vuex.Store({
      modules: {
        ui: {
          namespaced: true,
          state: {
            isTermsOfUseVisible: true
          },
          mutations: { ...uiMutations }
        }
      }
    })
  })

  test('render the terms of use', () => {
    const wrapper = createTheImageViewer()

    expect(wrapper.element).toMatchSnapshot()
  })

  test('closing by clicking a "close" button', () => {
    const wrapper = createTheImageViewer()

    const closeButton = wrapper.find('.terms-of-use__close')
    closeButton.trigger('click')
    expect(store.state.ui.isTermsOfUseVisible).toBe(false)
  })

  test('closing by clicking a link', () => {
    const wrapper = createTheImageViewer()

    const link = wrapper.find('.terms-of-use__link')
    link.trigger('click')
    expect(store.state.ui.isTermsOfUseVisible).toBe(false)
  })

  test('closing by clicking a "accept" button', () => {
    const wrapper = createTheImageViewer()

    const btn = wrapper.find('.terms-of-use__btn-accept')
    btn.trigger('click')
    expect(store.state.ui.isTermsOfUseVisible).toBe(false)
  })
})
