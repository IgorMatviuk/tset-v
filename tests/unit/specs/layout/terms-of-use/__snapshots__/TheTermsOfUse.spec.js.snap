// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TheImageViewer render the terms of use 1`] = `
<transition-stub
  name="fade"
>
  <div
    class="terms-of-use"
  >
    <div
      class="terms-of-use__content"
    >
      <div
        class="terms-of-use__text"
      >
        
        termsOfUse.content
        
        <nuxtlink-stub
          class="terms-of-use__link"
          to="privacy-policy"
        >
          
          termsOfUse.linkText
        
        </nuxtlink-stub>
      </div>
       
      <button
        class="btn terms-of-use__btn-accept btn-primary"
        type="button"
      >
        
        buttons.accept
      
      </button>
       
      <button
        class="terms-of-use__close"
        type="button"
      >
        <svg-icon-stub
          class="terms-of-use__icon"
          height="14"
          name="close"
          width="14"
        />
      </button>
    </div>
  </div>
</transition-stub>
`;
